electronic-tariff-file-rotations/main.go

```diff
@@ -26,3 +40,115 @@ func main() {
 		handler(nil)
 	}
 }
+
+func initializeEnvironment() {
+	if _, err := os.Stat(".env"); err == nil {
+		err := godotenv.Load()
+
+		if err != nil {
+			log.Fatal("Error loading .env file")
+		}
+	}
+
+	for key, defaultValue := range defaultConfig {
+		if val, exists := os.LookupEnv(key); !exists || val == "" {
+			err := os.Setenv(key, defaultValue)
+
+			if err != nil {
+				log.Fatal("Error occurred while setting environment variable")
+			}
+		}
+	}
+}
+
+func getAWSSession() *session.Session {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String("eu-west-2"),
+	})
+
+	if err != nil {
+		log.Fatal("Error while creating AWS session.")
+	}
+
+	return sess
+}
+
+func isDeletionCandidate(file S3File) bool {
+	const layout = "2006-01-02"
+	deletionDays, _ := strconv.ParseInt(os.Getenv("DELETION_CANDIDATE_DAYS"), 10, 64)
+
+	curtime := time.Now()
+	date, _ := time.Parse(layout, file.age)
+	dayDiff := int64((curtime.Sub(date)).Hours() / 24)
+
+	if dayDiff >= deletionDays {
+		return true
+	} else {
+		return false
+	}
+}
+
+func handler(event *LambdaEvent) {
+	initializeEnvironment()
+
+	sess := getAWSSession()
+	s3svc := s3.New(sess)
+
+	var deletionList []S3File
+
+	debug := os.Getenv("DEBUG") == "true"
+
+	resp, err := s3svc.ListObjectsV2(&s3.ListObjectsV2Input{
+		Bucket: aws.String(os.Getenv("ETF_BUCKET")),
+		Prefix: aws.String(os.Getenv("S3_PREFIX")),
+	})
+
+	if err != nil {
+		log.Println("Error getting bucket files.")
+		log.Fatal(err)
+	}
+
+	for _, item := range resp.Contents {
+		file := S3File{&s3.ObjectIdentifier{Key: item.Key}, item.LastModified.Format("2006-01-02")}
+
+		if (strings.Contains(*item.Key, os.Getenv("S3_SEARCH_TERM"))) && isDeletionCandidate(file) {
+			deletionList = append(deletionList, file)
+
+			if debug {
+				log.Printf("Deletion candidate found: %s\n", file.key)
+			}
+		}
+	}
+
+	if len(deletionList) != 0 {
+		deleteKeys := make([]*s3.ObjectIdentifier, len(deletionList))
+
+		i := 0
+		for _, file := range deletionList {
+			deleteKeys[i] = file.key
+			i++
+		}
+
+		if debug {
+			log.Println("Debug mode active, forgoing file deletion.")
+		} else {
+			_, err := s3svc.DeleteObjects(&s3.DeleteObjectsInput{
+				Bucket: aws.String(os.Getenv("ETF_BUCKET")),
+				Delete: &s3.Delete{
+					Objects: deleteKeys,
+					Quiet:   aws.Bool(false),
+				},
+			})
+
+			if err != nil {
+				log.Println("Error deleting files.")
+				log.Fatal(err)
+			}
+		}
+
+	} else {
+		log.Printf("No candidates for deletion. Exiting!\n")
+	}
+
+	os.Exit(0)
```

Comment:
I think this is actually implicitly what a go program will do when it exits execution

Reply:
That makes sense. I'll drop this in a commit